### PR TITLE
update rest api docs

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
#### Description of the pull request (what is new / what has changed)
update Rest API docs

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [x] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 1]
```diff
@@ -479,3 +515,249 @@ Transaction status retrieved successfully.
...
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## <span class="badge badge-primary">GET</span> **Get Transactions Pool for a Sender**
+
+`https://gateway.elrond.com/transaction/pool?by-sender=:sender:`
+
+This endpoint allows one to fetch all the transactions of a sender from the transactions pool.
+
+### Example with no field
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:591]: I would duplicate L591-L682 inside `Get Transactions Pool` as well as fields apply in both cases

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 2]
```diff
@@ -70,15 +73,19 @@ Nonce successfully retrieved.
...
+  "code": "successful"
 }
 ```
 
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 ## <span class="badge badge-primary">GET</span> **Get Address Balance**
 
-https://gateway.elrond.com**/address/:bech32Address/balance**
+`https://gateway.elrond.com**/address/:bech32Address/balance**`
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/addresses.md:88]: are the ** ** intended?

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 3]
```diff
@@ -119,60 +130,98 @@ This endpoint allows one to retrieve the latest 20 Transactions sent from an Add
...
+  "error": "",
+  "code": "successful"
+}
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## <span class="badge badge-primary">GET</span> **Get Address Transactions**
+
+`https://gateway.elrond.com**/address/:bech32Address/transactions**`
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/addresses.md:156]: are the ** ** intended?

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 4]
```diff
@@ -188,7 +237,7 @@ This endpoint requires the presence of an Elasticsearch instance (populated thro
 
 ## <span class="badge badge-primary">GET</span> **Get Storage Value for Address**
 
-https://gateway.elrond.com**/address/:bech32Address/key/:key**
+`https://gateway.elrond.com**/address/:bech32Address/key/:key**`
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/addresses.md:240]: are the ** ** intended?

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 5]
```diff
@@ -213,15 +262,19 @@ Value (hex-encoded) successfully retrieved.
...
+    "code": "successful"
 }
 ```
 
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 ## <span class="badge badge-primary">GET</span> **Get all storage for Address**
 
-https://gateway.elrond.com**/address/:bech32Address/keys**
+`https://gateway.elrond.com**/address/:bech32Address/keys**`
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/addresses.md:277]: same here ** **

File: docs/sdk-and-tools/rest-api/gateway-overview.md [hunk 6]
```diff
@@ -11,10 +11,10 @@ The difference is that `gateway` is Elrond's infrastructure backed by the [elron
 
 ## **Proxy API vs. Observer API**
 
-Both the **Elrond Proxy** and the **Node (Observer)** are designed to expose the same HTTP endpoints, though **this doesn't always hold due to architectural constraints**. When describing each HTTP endpoint on the following pages, if there is any discrepancy or mismatch between the _Proxy endpoint_ and the _Observer endpoint_, this will be captured in a note as the one below:
+Both the **Elrond Proxy** and the **Node (Observer)** are designed to expose (almost) the same HTTP endpoints, though **this doesn't always hold due to architectural constraints**. When describing each HTTP endpoint on the following pages, if there is any discrepancy or mismatch between the _Proxy endpoint_ and the _Observer endpoint_, this will be captured in a note as the one below:
 
 :::important
-`**example/endpoint**` is only available on the Elrond Proxy.
+Proxy/Gateway endpoints are referred as `https://gateway.elrond.com/....`, while node endpoints are referred as `http://localhost:8080/....`.
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/gateway-overview.md:17]: are ::: intended before important? same at the end of `network.md` file

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 1 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:591]: done

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 7]
```diff
@@ -29,26 +29,29 @@ Address information successfully retrieved.
...
+    "code": "success"
 }
 ```
 
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 ## <span class="badge badge-primary">GET</span> **Get Address Nonce**
 
-https://gateway.elrond.com**/address/:bech32Address/nonce**
+`https://gateway.elrond.com**/address/:bech32Address/nonce**`
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:54]: fixed

File: docs/sdk-and-tools/rest-api/addresses.md (hunk 2 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:88]: fixed

File: docs/sdk-and-tools/rest-api/addresses.md [hunk 8]
```diff
@@ -100,17 +107,21 @@ Balance successfully retrieved.
...
 }
 ```
 
 <!--END_DOCUSAURUS_CODE_TABS-->
 
-## <span class="badge badge-primary">GET</span> **Get Address Transactions**
+## <span class="badge badge-primary">GET</span> **Get Address Username (herotag)**
 
-https://gateway.elrond.com**/address/:bech32Address/transactions**
+`https://gateway.elrond.com**/address/:bech32Address/username**`
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:122]: fixed

File: docs/sdk-and-tools/rest-api/addresses.md (hunk 3 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:156]: fixed

File: docs/sdk-and-tools/rest-api/addresses.md (hunk 4 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:240]: fixed

File: docs/sdk-and-tools/rest-api/addresses.md (hunk 5 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/addresses.md:277]: fixed

File: docs/sdk-and-tools/rest-api/gateway-overview.md (hunk 6 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/gateway-overview.md:17]: yes, they are intended

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 9]
```diff
@@ -17,21 +17,21 @@ This endpoint allows one to send a signed Transaction to the Blockchain.
...
+| gasPrice         | <span class="text-danger">REQUIRED</span> | `number` | The desired Gas Price (per Gas Unit).                                            |
+| gasLimit         | <span class="text-danger">REQUIRED</span> | `number` | The maximum amount of Gas Units to consume.                                      |
+| data             | <span class="text-normal">OPTIONAL</span> | `string` | The base64 string representation of the Transaction's message (data).            |
+| signature        | <span class="text-danger">REQUIRED</span> | `string` | The Signature (hex-encoded) of the Transaction.                                  |
+| chainID          | <span class="text-danger">REQUIRED</span> | `string` | The Chain identifier.                                                            |
+| version          | <span class="text-danger">REQUIRED</span> | `number` | The Version of the Transaction (e.g. 1).                                         |
+| options          | <span class="text-normal">OPTIONAL</span> | `number` | The Options of the Transaction (e.g. 1).                                         |
 
 <!--Response-->
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:37]: done

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 10]
```diff
@@ -479,3 +515,249 @@ Transaction status retrieved successfully.
...
+As seen above, if the `fields` item is empty, only the transaction hash will be displayed.
+
+Example request with fields:
+
+`https://gateway.elrond.com/transaction/pool?by-sender=erd1at9...&fields=sender,receiver,value`
+
+All possible values are:
+
+- hash           
+- nonce
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md]: done

**Review by ssd04 — APPROVED** [review]

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 11]
```diff
@@ -479,3 +521,333 @@ Transaction status retrieved successfully.
...
+
+### Using custom fields
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Request-->
+
+Query Parameters
+
+| Param     | Required                                  | Type     | Description                          |
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:591]: maybe remove the table here as it is described above? or delete it from above

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 12]
```diff
@@ -479,3 +521,333 @@ Transaction status retrieved successfully.
...
+|-----------|-------------------------------------------|----------|--------------------------------------|
+| fields    | <span class="text-normal">OPTIONAL</span> | `string` | A list of the fields to be included. |
+
+s seen above, if the `fields` item is empty, only the transaction hash will be displayed.
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:595]:
```suggestion
As seen above, if the `fields` item is empty, only the transaction hash will be displayed.
```

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 13]
```diff
@@ -479,3 +521,333 @@ Transaction status retrieved successfully.
...
+
+Example request with fields:
+
+`https://gateway.elrond.com/transaction/pool?fields=sender,receiver,value`
+
+All possible values for fields item are:
+
+- hash
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:603]: hash is already default, nothing happens if added to fields

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 14]
```diff
@@ -479,3 +521,333 @@ Transaction status retrieved successfully.
...
+- nonce
+- sender
+- receiver
+- gaslimit
+- gasprice
+- receiverusername
+- data
+- value
+-
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:612]:
```suggestion

```

File: docs/sdk-and-tools/rest-api/transactions.md [hunk 15]
```diff
@@ -479,3 +521,333 @@ Transaction status retrieved successfully.
...
+
+As seen above, if the `fields` item is empty, only the transaction hash will be displayed.
+
+Example request with fields:
+
+`https://gateway.elrond.com/transaction/pool?by-sender=erd1at9...&fields=sender,receiver,value`
+
+All possible values for fields item are:
+
+- hash
```
Comment (sstanculeanu) [inline docs/sdk-and-tools/rest-api/transactions.md:735]: same note here, hash is default and nothing happens if it is added to fields

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 12 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:595]: right, fixed

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 13 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:603]: I know, but it's more clear this way

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 14 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:612]: fixed

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 15 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:735]: would leave it as it is

File: docs/sdk-and-tools/rest-api/transactions.md (hunk 11 above)
Comment (bogdan-rosianu) [reply docs/sdk-and-tools/rest-api/transactions.md:591]: right. fixed

**Review by sstanculeanu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**